packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -116,14 +116,18 @@ class EnrollVerbHandler extends AbstractVerbHandler {
       throw AtThrottleLimitExceeded(
           'Enrollment requests have exceeded the limit within the specified time frame');
     }
-    if (!atConnection.getMetaData().isAuthenticated) {
-      var otp = enrollParams.otp;
-      if (otp == null ||
-          (await OtpVerbHandler.cache.get(otp.toString()) == null)) {
+
+    // OTP is sent only in enrollment request which is submitted on
+    // unauthenticated connection.
+    if (atConnection.getMetaData().isAuthenticated == false) {
+      Response otpResponse = await OtpVerbHandler(keyStore)
+          .processInternal('otp:validate:${enrollParams.otp}', atConnection);
```

Comment:
I don't like the idea of verb handlers using internal functionality of other verb handlers. Is there a better / cleaner way of making that validation functionality more explicitly available?

Reply:
Moved the "isOTPValid" method to the abstract_verb_handler.dart which wil be invoked by otp_verb_handler and enroll_verb_handler.